cluster/nats_rpc_common.go

```diff
@@ -34,8 +34,8 @@ func getChannel(serverType, serverID string) string {
 func setupNatsConn(connectString string, appDieChan chan bool, options ...nats.Option) (*nats.Conn, error) {
 	natsOptions := append(
 		options,
-		nats.DisconnectHandler(func(_ *nats.Conn) {
-			logger.Log.Warn("disconnected from nats!")
+		nats.DisconnectErrHandler(func(_ *nats.Conn, err error) {
+			logger.Log.Warnf("disconnected from nats! Reason: %q\n", err)
```

Comment:
Why we're using `%q`, this would translate to a single-quoted character, don't we want to go with `%s`?

Reply:
%q will quote each element of a slice of strings not a single character
i used %q because that's what was used in the nats.go examples